pkg/controller/download_request_controller_test.go

```diff
@@ -18,310 +18,256 @@ package controller
-			restore:         builder.ForRestore(velerov1api.DefaultNamespace, "a-backup-20170912150214").Phase(velerov1api.RestorePhaseCompleted).Backup("a-backup").Result(),
-			backup:          defaultBackup(),
-			backupLocation:  newBackupLocation("a-location", "a-provider", "a-bucket"),
-			expectGetsURL:   true,
-		},
-		{
-			name:            "restore results request with phase 'New' gets a url",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseNew, velerov1api.DownloadTargetKindRestoreResults, "a-backup-20170912150214"),
-			restore:         builder.ForRestore(velerov1api.DefaultNamespace, "a-backup-20170912150214").Phase(velerov1api.RestorePhaseCompleted).Backup("a-backup").Result(),
-			backup:          defaultBackup(),
-			backupLocation:  newBackupLocation("a-location", "a-provider", "a-bucket"),
-			expectGetsURL:   true,
-		},
-		{
-			name:            "request with phase 'Processed' is not deleted if not expired",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseProcessed, velerov1api.DownloadTargetKindBackupLog, "a-backup-20170912150214"),
-			backup:          defaultBackup(),
-		},
-		{
-			name:            "request with phase 'Processed' is deleted if expired",
-			downloadRequest: newDownloadRequest(velerov1api.DownloadRequestPhaseProcessed, velerov1api.DownloadTargetKindBackupLog, "a-backup-20170912150214"),
-			backup:          defaultBackup(),
-			expired:         true,
-		},
-	}
+		var (
+			pluginManager = &pluginmocks.Manager{}
+			backupStores  = make(map[string]*persistencemocks.BackupStore)
+		)
+		pluginManager.On("CleanupClients").Return(nil)
 
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			var fakeClient client.Client
-			if tc.backupLocation != nil {
-				fakeClient = newFakeClient(t, tc.backupLocation)
-			} else {
-				fakeClient = newFakeClient(t)
-			}
+		for _, tc := range tests {
+			// now will be used to set the fake clock's time; capture
+			// it here so it can be referenced in the test case defs.
+			now, err := time.Parse(time.RFC1123, time.RFC1123)
+			Expect(err).To(BeNil())
+			now = now.Local()
 
-			harness := newDownloadRequestTestHarness(t, fakeClient)
+			fakeClient := fake.NewFakeClientWithScheme(scheme.Scheme)
 
-			// set up test case data
+			rClock := clock.NewFakeClock(now)
 
-			// Set .status.expiration properly for processed requests. Since "expired" is relative to the controller's
-			// clock time, it's easier to do this here than as part of the test case definitions.
-			if tc.downloadRequest != nil && tc.downloadRequest.Status.Phase == velerov1api.DownloadRequestPhaseProcessed {
-				if tc.expired {
-					tc.downloadRequest.Status.Expiration = &metav1.Time{Time: harness.controller.clock.Now().Add(-1 * time.Minute)}
-				} else {
-					tc.downloadRequest.Status.Expiration = &metav1.Time{Time: harness.controller.clock.Now().Add(time.Minute)}
+			if tc.downloadRequest != nil {
```

Comment:
In all the test cases, `tc.downloadRequest` is not nil and it seems like it's necessary that it isn't nil. If you wanted to be really strict, you could fail the test early if it's nil?

Reply:
Yes, good point. Fixed.